spec.bs

```diff
@@ -1604,7 +1604,68 @@ For now, see [https://github.com/wicg/observable#operators](https://github.com/w
   The <dfn for=Observable method><code>reduce(|reducer|, |initialValue|, |options|)</code></dfn>
   method steps are:
 
-    1. <span class=XXX>TODO: Spec this and use |reducer|, |initialValue|, and |options|.</span>
+    1. Let |p| [=a new promise=].
+
+    1. Let |controller| be a [=new=] {{AbortController}}.
+
+    1. Let |internal options| be a new {{SubscribeOptions}} whose {{SubscribeOptions/signal}} is the
+       result of [=creating a dependent abort signal=] from the list
+       «|controller|'s [=AbortController/signal=], |options|'s
+       {{SubscribeOptions/signal}} if non-null», using {{AbortSignal}}, and the [=current realm=].
+
+    1. If |internal options|'s {{SubscribeOptions/signal}} is [=AbortSignal/aborted=], then:
+
+       1. [=Reject=] |p| with |internal options|'s {{SubscribeOptions/signal}}'s
+          [=AbortSignal/abort reason=].
+
+       1. Return |p|.
+
+    1. [=AbortSignal/add|Add the following abort algorithm=] to |internal options|'s
+       {{SubscribeOptions/signal}}:
+
+       1. [=Reject=] |p| with |internal options|'s {{SubscribeOptions/signal}}'s [=AbortSignal/abort
+          reason=].
+
+    1. Let |idx| be an {{unsigned long long}}, initially 0.
+
+    1. Let |accumulator| be |initialValue| if it is given, and the string "<code>unset</code>"
```

Comment:
Using a string here is a hazard with user code as this string could well be passed as the initial value by user code, or returned from the reducer.

Reply:
Ah yeah, for sure. As I was implementing this "unset" was really just an internal-only value, so I'll ensure that the alternate form of this variable is something like that too, instead of something that developers can literally pass in heh. Whoops.

Reply:
I've decided to use the uninitialized state of `|accumulator|` as a signal for whether `|initialValue|` was passed in initially. That saves us spec-wise from having to maintain a separate variable of `|accumulatorHasValue|` or something like that, yet has the same semantics.